src/mmw/js/src/draw/views.js

```diff
@@ -81,7 +81,10 @@ function validateRwdShape(result) {
             d.reject('Unfortunately, the watershed generated at this ' +
                      'location is not available for analysis');
         }
-        validateShape(result.watershed)
+        // Validate the shape as a multipolygon -- this is
+        // how it'll be displayed on the map, and sent to
+        // analyze
+        validateShape(coreUtils.toMultiPolygon(result.watershed))
```

Comment:
Is the result of this that RWD output (like the Gibraltar example I gave) can be analyzed via the app, but not the API? That's what I would expect, since there is this conversion which is neccessary.  However, I can take the results from the watershed API and plug that directly into the soils API and get a successful result.  Can you explain what I'm not getting?

```
{"location":[40.28895415740959,-75.86832046508789],"snappingOn":true,"dataSource":"drb"}
```

Reply:
I can't explain it fully, but my hunch is the holes in the polygon are ignored once it's contained in a multipolygon. The same conversion happens on the backend here (https://github.com/WikiWatershed/model-my-watershed/pull/2487/files#diff-10de29d8b4c29bab0a1f77cc894dc45cR58) before validation.

I'll dig into this a little.

Reply:
What I think I've found is `turfKinks` and `shapely.is_valid` don't check a certain sort of validity deeply. When we pass a `Polygon` they check if the shape intersects, and if the rings intersect each other. When we pass a `MultiPolygon`, they check if any of the `Polygon`s intersect each other, or intersect themselves , but not if the `Polygon`s have intersecting rings.

So, in one sense, forcing these shapes into `MultiPolygon`s is just muffling a legitimate error. In another sense, the error isn't all that legitimate because the geoproccessing seems like it can handle shapes with these kinds of self-intersection, namely rings that intersect borders:

(The mask in the app hides these rings, but here's one thrown into geojson.io from the same gibraltar shape)
![screen shot 2017-11-07 at 4 04 32 pm](https://user-images.githubusercontent.com/7633670/32519233-0cf1e45e-c3da-11e7-9ba6-8a2858a35999.png)

It's maybe worth noting that the muffling was happening with RWD shapes before this PR. `result.watershed` was made a `MultiPolygon` before it got to `validateShape`.

If we don't want to keep the muffling, some other ideas I have for handling the ringed shapes are:
- Remove rings either in RWD or on the frontend
- During frontend validation, validate only the outer ring of a polygon
- During frontend validation, validate each ring of a polygon, but not the polygon as a whole 

Any thoughts on what would be cleanest?

Reply:
@arottersman Sorry, I didn't catch this message.  Yes, this is kind of what I expected, that the validation is too strong compared to what can actually be run through the processor.  Given your options to move forward, I'd have these recommendations:
* As a priority, try not to alter the shape of the watershed
* Same behavior between app and API
* Don't prevent shapes which can be analyzed from being so, even if that means some validations are ambiguous and we attempt to process polygons which can't be handled well
* In the above case, we catch and report the invalid input error as before, it'll presumably just take a bit longer to get to the user.

Do those guidelines help clarify an approach you'd prefer?